sts-mock/src/token/tokenHandler.test.ts

```diff
@@ -1,41 +1,184 @@
+        );
+      });
+    });
+  });
+
+  describe("Key Retriever", () => {
+    describe("Given an error happens trying to get the signing key from S3", () => {
+      it("Returns 500 Server Error", async () => {
+        dependencies.keyRetriever = () => new MockKeyRetrieverErrorResult();
+
+        const result = await lambdaHandlerConstructor(
+          dependencies,
+          event,
+          buildLambdaContext(),
+        );
+
+        expect(mockLogger.getLogMessages()[0].logMessage.message).toStrictEqual(
+          "STARTED",
+        );
+        expect(mockLogger.getLogMessages()[1].logMessage.message).toStrictEqual(
+          "INTERNAL_SERVER_ERROR",
+        );
+        expect(mockLogger.getLogMessages()[1].data).toStrictEqual({
+          errorMessage: "Some S3 error",
+        });
+        expect(result.statusCode).toStrictEqual(500);
+        expect(result.body).toStrictEqual(
+          '{"error":"server_error","error_description":"Server Error"}',
+        );
+      });
+    });
+  });
+
+  describe("Token Signer", () => {
+    describe("Given an error happens trying to crate a signed token", () => {
+      it("Returns 500 Server Error", async () => {
+        dependencies.tokenSigner = () => new MockTokenSignerErrorResult();
+
+        const result = await lambdaHandlerConstructor(
+          dependencies,
+          event,
+          buildLambdaContext(),
+        );
+
+        expect(mockLogger.getLogMessages()[0].logMessage.message).toStrictEqual(
+          "STARTED",
+        );
+        expect(mockLogger.getLogMessages()[1].logMessage.message).toStrictEqual(
+          "INTERNAL_SERVER_ERROR",
+        );
+        expect(mockLogger.getLogMessages()[1].data).toStrictEqual({
+          errorMessage: "Some signing error",
+        });
+        expect(result.statusCode).toStrictEqual(500);
+        expect(result.body).toStrictEqual(
+          '{"error":"server_error","error_description":"Server Error"}',
+        );
+      });
+    });
+  });
+
+  describe("Issue service token", () => {
+    describe("Given the request is valid and the service token is generated", () => {
+      it("Returns 200 and the service token", async () => {
+        const result = await lambdaHandlerConstructor(
+          dependencies,
+          event,
+          buildLambdaContext(),
+        );
+
+        expect(mockLogger.getLogMessages()[0].logMessage.message).toStrictEqual(
+          "STARTED",
+        );
+        expect(mockLogger.getLogMessages()[1].logMessage.message).toStrictEqual(
+          "COMPLETED",
+        );
+        expect(result).toStrictEqual({
+          headers: {
+            "Content-Type": "application/json",
+            "Cache-Control": "no-store",
+            Pragma: "no-cache",
+          },
+          statusCode: 200,
+          body: JSON.stringify({
+            access_token: "mockServiceToken",
+            token_type: "Bearer",
+            expires_in: 180,
```

Comment:
Is this expiry value corect?

Reply:
As per https://govukverify.atlassian.net/wiki/spaces/DCMAW/pages/3844964353/Strategic+App+App+calls+a+protected+service#Example-Response and https://github.com/govuk-one-login/sts-back/blob/cc84af37f4354458a1b1ac12c8c381df5710ad2b/backend-api/template.yaml#L2912, it is correct, unless I'm not looking in the right place